### DISCUSSION TITLE
How to integrate StyleX in RsBuild

### DISCUSSION THREAD

**Discussion by longhtcoin98** [facebook/stylex#568] [Q&A] [answered]
I'm using RsBuild (https://rsbuild.dev/) to create React app and integrate StyleX.
It's working in Dev mode, but it does not create CSS file in Production mode.
Here my configs
```
import { defineConfig } from '@rsbuild/core';
import { pluginReact } from '@rsbuild/plugin-react';
import { pluginBabel } from '@rsbuild/plugin-babel';
import styleXPlugin from '@stylexjs/babel-plugin';

export default defineConfig({
  plugins: [
    pluginBabel({
      babelLoaderOptions: (config) => {
        config.plugins ||= []
        config.plugins.push(
          [
              styleXPlugin,
            {
              dev: false, // = Production
              test: false,
              unstable_moduleResolution: {
                type: 'commonJS',
                rootDir: __dirname,
              },
            },
          ]
        )
      },
    }),
    pluginReact()
  ],
});
```

So, I don't know what happened, please help me. Thank you

**Comment by nedjulius** [comment]
hey, @eryue0220 has created an `unplugin` StyleX plugin -- you can try to use it in your rspack config https://github.com/eryue0220/unplugin-stylex

**Reply by longhtcoin98** [reply]
![Screenshot 2024-05-09 214328](https://github.com/facebook/stylex/assets/150644299/6c2ef199-b869-4262-aa42-89498464385e)
![Screenshot 2024-05-09 214416](https://github.com/facebook/stylex/assets/150644299/856a28d8-6283-4d30-8bc3-422f30b4fe1a)

hi @nedjulius, it doesn't compatible with RsBuild.

**Reply by eryue0220** [reply]
Hi @longhtcoin98, unplugin-stylex author here. Could you open an issue under the [unplugin-stylex](https://github.com/eryue0220/unplugin-stylex/issues) and provide a minimal reproduction repo or description about the detail? I'll try on it.

**Comment by longht021189** [comment]
> Hi @longhtcoin98, unplugin-stylex author here. Could you open an issue under the [unplugin-stylex](https://github.com/eryue0220/unplugin-stylex/issues) and provide a minimal reproduction repo or description about the detail? I'll try on it.

Hi @eryue0220, I provided a minimal reproduction repo [here](https://github.com/eryue0220/unplugin-stylex/issues/37), Please help me check it. Thank you.

**Reply by eryue0220** [reply]
Thanks, I'll check it in a moment.

**Comment by eryue0220** [comment, marked as answer]
Thanks for the feed back, and please upgrade `unplugin-stylex` to `v0.3.1`, I've fixed this bug.

**Reply by longht021189** [reply]
Thank you, It's fixed in `unplugin-stylex v0.3.1`

**Comment by nmn** [comment]
Thanks for the great work @eryue0220!
